### DISCUSSION TITLE
Updating zigbee firmware fails on 9.2

### DISCUSSION THREAD

**Discussion by yvesdm3000** [arendst/Tasmota#10343] [Q&A] [answered]
Hello

I am setting up a second sonoff zbbridge. I flashed it with the latest tasmota 9.2 fw version for zigbee but the last step fails: I need to upload the zigbee firmware like I did on my first device but fails with "invalid file signature. This is my log:

00:00:00 CFG: Loaded from flash at FB, Count 40
00:00:00 QPC: Count 1
00:00:00 SNS: Hardware Serial
00:00:00 SRC: Restart
00:00:00 Project tasmota Tasmota Version 9.2.0(zbbridge)-2_7_4_9(2020-12-21T15:06:32)
00:00:00 ZIG: Resetting EZSP device
00:00:00 WIF: Attempting connection...
00:00:00 WIF: Connecting to AP2 RODE_INTERIOR Channel 3 BSSId 00:23:69:2A:22:44 in mode 11n as ZigbeeBridge2...
00:00:01 WIF: Connected
00:00:02 HTP: Web server active on ZigbeeBridge2 with IP address 172.16.0.133
07:47:27 RTC: UTC 2021-01-01T06:47:27, DST 2021-03-28T02:00:00, STD 2021-10-31T03:00:00
07:47:28 RSL: tele/tasmota_E30431/INFO1 = {"Module":"Sonoff ZBBridge","Version":"9.2.0(zbbridge)","FallbackTopic":"cmnd/DVES_E30431_fb/","GroupTopic":"cmnd/tasmotas/"}
07:47:28 RSL: tele/tasmota_E30431/INFO2 = {"WebServerMode":"Admin","Hostname":"ZigbeeBridge2","IPAddress":"172.16.0.133"}
07:47:28 RSL: tele/tasmota_E30431/INFO3 = {"RestartReason":"Software/System restart"}
07:47:31 QPC: Reset
07:47:31 HTP: Main Menu
07:47:32 RSL: tele/tasmota_E30431/STATE = {"Time":"2021-01-01T07:47:32","Uptime":"0T00:00:09","UptimeSec":9,"Vcc":3.440,"Heap":28,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"Wifi":{"AP":2,"SSId":"RODE_INTERIOR","BSSId":"00:23:69:2A:22:44","Channel":3,"RSSI":14,"Signal":-93,"LinkCount":1,"Downtime":"0T00:00:03"}}
07:47:33 HTP: Firmware Upgrade
07:47:33 APP: Boot Count 7
07:47:33 CFG: Saved to flash at FA, Count 41, Bytes 4096
07:47:39 CFG: Saved to flash at FB, Count 42, Bytes 4096
07:47:39 UPL: File ncp-uart-sw_6.7.6_115200.ota ...
07:47:43 HTP: Upload done
07:47:43 ZIG: Resetting EZSP device
07:47:44 ZbEZSPReceived: time-out, discarding FE0D0180020000000001010474, 1
07:47:48 ZIG: timeout, goto label 99
07:47:48 RSL: tele/tasmota_E30431/RESULT = {"ZbState":{"Status":99,"Message":"Abort"}}
07:47:48 ZIG: Abort
07:47:48 ZIG: Stopping (99)
07:52:35 RSL: tele/tasmota_E30431/STATE = {"Time":"2021-01-01T07:52:35","Uptime":"0T00:05:12","UptimeSec":312,"Vcc":3.438,"Heap":28,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"Wifi":{"AP":2,"SSId":"RODE_INTERIOR","BSSId":"00:23:69:2A:22:44","Channel":3,"RSSI":16,"Signal":-92,"LinkCount":1,"Downtime":"0T00:00:03"}}
07:55:21 HTP: Main Menu
07:55:23 HTP: Console
07:57:35 RSL: tele/tasmota_E30431/STATE = {"Time":"2021-01-01T07:57:35","Uptime":"0T00:10:12","UptimeSec":612,"Vcc":3.438,"Heap":27,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"Wifi":{"AP":2,"SSId":"RODE_INTERIOR","BSSId":"00:23:69:2A:22:44","Channel":3,"RSSI":14,"Signal":-93,"LinkCount":1,"Downtime":"0T00:00:03"}}

Am I doing something wrong or should I try downgrading to 9.1 like my other device?

Yves

**Comment by Jason2866** [comment]
Have you verified that the downloaded NCP file is not damaged?

**Reply by yvesdm3000** [reply]
Feedback from Discord:

MD5sum:  50585b8b113a73fa9159706680577c83

Should be correct !

I downgraded tasmota to 9.1, did a "Reset 1" and then it allowed me to upload the firmware.

After that I upgraded again to tasmota 9.2 and the device is working now. It looks to me there is some issue with 9.2 and uploading this firmware.

Yves

**Comment by robitmoh** [comment]
Similar problem
I have working zbride in ZHA mode with tasmota 8.5 .
Take ota upgrade  ->  upgraded to 9.2.
The Zigbee module is not started .
I try to upgrade more time a zigbee fw but always "invalid file signature" coming.

Try downgrade to 9.1 and Zha mode is starting, but can't upgrade zigbee fw,   "Magic byte...." error
Take a reset 4 
Successfully upgrade the zigbee fw.

Take ota upgrade to 9.2
The upgrade successfully but can't select the Sonoff ZbridgeModule (75)
Try the manually upgrade, with downloaded 9.2 fw  file, the upgrade successfully but can't select the Sonoff ZbridgeModule (75)
Take a reset 5 and after reboot, the  Sonoff ZbridgeModule (75) already selected 

The first see everything is working

**Comment by s-hadinger** [comment]
The fact that you can't select Module 75 seems to indicate that you had the wrong firmware but your confirmed that it worked afterwards which shows it was the right firmware after all. I don't see how a reset 1 can make it reappear.

**Reply by s-hadinger** [reply]
@arendst Is there anything module related that comes to your mind?

**Comment by arendst** [comment, marked as answer]
If user upgrades with a different binary than ``tasmota-zbbridge.bin`` he won't see module 75. This module is only visible in binary ``tasmota-zbbridge.bin``
